src/rocprof_compute_soc/soc_base.py

```diff
@@ -299,6 +324,45 @@ def add(self, counter) -> bool:
         return self.blocks[block].add(counter)
 
 
+def getblock(counter):
+    return counter.split("_")[0]
+
+
+# Set with limited size
+class LimitedSet:
+    def __init__(self, maxsize) -> None:
+        self.avail = maxsize
+        self.elements = []
+
+    def add(self, e) -> None:
+        if e in self.elements:
+            return True
+        elif self.avail <= 0:
+            return False
+
+        self.avail -= 1
+        self.elements.append(e)
+
+        return True
+
+
+# Represents a file that lists PMC counters. Number of counters for each
+# block limited according to perfmon config.
+class CounterFile:
+    def __init__(self, name, perfmon_config) -> None:
+        self.file_name = name
+        self.blocks = {b: LimitedSet(v) for b, v in perfmon_config.items()}
+
+    def add(self, counter) -> bool:
+        block = getblock(counter)
+
+        # SQ and SQC belong to the same IP block
+        if block == "SQC":
+            block = "SQ"
+
+        return self.blocks[block].add(counter)
+
+
```

Comment:
Where are these changes coming from? Is this within the scope of this PR?

Reply:
Looks like it was merged to resolve conflict. Removing this content in this PR.